src/common/services/user/index.js

```diff
@@ -95,10 +96,14 @@ function* login({ payload }) {
 }
 
 function* relogin() {
-  try {
-    const resp = yield call(api.relogin);
-    yield put(loginActions.success(resp.data));
-  } catch {
+  if (!(yield select(selectIsLoggingOut)) && (yield select(selectIsLoggedIn))) {
+    try {
+      const resp = yield call(api.relogin);
+      yield put(loginActions.success(resp.data));
+    } catch {
+      yield put(loginActions.failure());
+    }
+  } else {
     yield put(loginActions.failure());
```

Comment:
why dispatching login.failure if user is nto loggedin? cant we do just nothing?

Reply:
Relogin sets `isAuthenticating` to true. We must disable it or the app is blocked.

Reply:
i dont mean `yield put(loginActions.failure());` in the `catch` but in the `else` block. Relogin is not being called if it falls into else block

Reply:
Yeah, I know. But the reducer is triggered when the action is dispatched. api.relogin has nothing to do with it so we must switch the state in the `else` block.

Reply:
oh that. in that case i think this is an ugly hack :D relogin reducer shouldnt be triggered at all and isAuthenticating shouldnt change because in reality, authenticating was never started